lib/Makefile.am

```diff
@@ -144,9 +144,9 @@ endif
 if HAVE_SPLINT
 check_SCRIPTS = run_splint.sh
 TESTS = $(check_SCRIPTS)
-# this is a hack because debian/ubuntu don't set the arch path
-# in splint.
-DEB_INCLUDES = -I/usr/include/x86_64-linux-gnu -I/usr/include/i386-linux-gnu
+# this is a hack because ubuntu(debian fixed in #675025) don't set the arch path
+# in splint (https://bugs.launchpad.net/ubuntu/+source/splint/+bug/1786658)
+DEB_INCLUDES = -I/usr/include/$$(dpkg-architecture -q DEB_HOST_GNU_TYPE)
```

Comment:
Not sure how versed you are with autotools, can give you more guidance
if desired (just ask), but what I'd like to see is:

1. `configure.ac`, when `splint` actually detected, will look for
   `dpkg-architecture` executable and sets the `HAVE_DPKG_ARCHITECTURE`
    automake conditional accordingly

2. in `lib/Makefile.am`, `DEB_INCLUDES` will either be set empty or
   to `-I/usr/include/$$($(DPKG_ARCHITECTURE) -q DEB_HOST_GNU_TYPE)`,
   almost like it is now with your current patch

IIUIC, you can still use, e.g.
```
make -C lib check DEB_INCLUDES=-I/usr/include...
```
if there's any pressing need to avoid having `dpkg-architecture` executable
installed.

And please, polish the non-code parts:
- space before `(`
- `don't` -> `doesn't`